pkg/apis/core/validation/validation.go

```diff
@@ -7175,7 +7175,7 @@ func ValidateSecurityContext(sc *core.SecurityContext, fldPath *field.Path, host
 		if err := ValidateProcMountType(fldPath.Child("procMount"), *sc.ProcMount); err != nil {
 			allErrs = append(allErrs, err)
 		}
-		if hostUsers && *sc.ProcMount == core.UnmaskedProcMount {
+		if utilfeature.DefaultFeatureGate.Enabled(features.UserNamespacesSupport) && hostUsers && *sc.ProcMount == core.UnmaskedProcMount {
```

Comment:
What happens if `UserNamespacesSupport` is disabled and `Unmasked` gets through here?

If this is ok, we should add test case where the feature is off to ensure this case is exercised properly.

Reply:
>What happens if UserNamespacesSupport is disabled and Unmasked gets through here?

The above condition is added to cover this use case. If you see the test failure, we are setting the `procMount` to `Unmasked` and the feature is disabled by default.

Reply:
oh got it now, `UnmaskedProcMount` is a valid value when `UserNamespacesSupport` is disabled since it will not be `true`  

We should add unit test `validation_test.go` to make sure this case is covered in unit tests?

Reply:
I think I confused you. Let me rephrase my previous reply. If you look at this test: [test link](https://github.com/kubernetes/kubernetes/blob/master/test/e2e/windows/security_context.go#L163) (which is failing at the moment), this test assigns `HostUsers` to false and sets `ProcMount` to Unmasked. However, the `HostUsers` field gets overridden when [this condition](https://github.com/kubernetes/kubernetes/blob/master/pkg/api/pod/util.go#L651) becomes true, which is the case right now (since we are not explicitly opting in for the `UserNamespacesSupport` feature in the test, that condition becomes true and `HostUsers` is nil). With that, the `hostUsers && *sc.ProcMount == core.UnmaskedProcMount` condition becomes true, and we are hitting this issue.

```sh
{ failed [FAILED] Pod "pod-77ced19a-4786-47e3-a166-cbffe9cb367c" is invalid: spec.containers[0].securityContext.procMount: Invalid value: "Unmasked": `hostUsers` must be false to use `Unmasked`
In [It] at: k8s.io/kubernetes/test/e2e/windows/security_context.go:181 @ 07/17/24 20:16:10.774
```

>oh got it now, UnmaskedProcMount is a valid value when UserNamespacesSupport is disabled since it will not be true

`UnmaskedProcMount` is valid when `UserNamespacesSupport` is enabled and `HostUsers` is set to `false`.

Reply:
I think [this](https://github.com/kubernetes/kubernetes/pull/126182/files#diff-7ee023b4401073292170227985c8ac98015db0c1980657c277a7cea011c24830R20889) should cover the unit test unless we want to have an exclusive test for it.